ios/demo/src/MapViewController.m

```diff
@@ -1,28 +1,74 @@
 //
-//  ViewController.m
+//  MapViewController.m
 //
 //  Created by Karim Naaji on 10/12/16.
 //  Copyright © 2016 Karim Naaji. All rights reserved.
 //
 
 #import "MapViewController.h"
 
+@implementation MapViewControllerDelegate
+
+- (void)mapView:(TGMapViewController *)mapView didLoadSceneAsync:(NSString *)scene
+{
+    NSLog(@"Did load scene async %@", scene);
+
+    TGGeoPoint newYork;
+    newYork.longitude = -74.00976419448854;
+    newYork.latitude = 40.70532700869127;
+
+    [mapView setZoom:16];
+    [mapView setPosition:newYork];
+}
+
+- (void)mapView:(TGMapViewController*)mapView didSelectFeatures:(NSDictionary *)features atScreenPosition:(CGPoint)position
+{
+    NSLog(@"Picked features:");
+
+    for (id key in features) {
+        NSLog(@"\t%@ -- %@", key, [features objectForKey:key]);
+    }
+}
+
+@end
+
+@implementation MapViewControllerRecognizerDelegate
+
+- (void)mapView:(TGMapViewController *)view recognizer:(UIGestureRecognizer *)recognizer didRecognizeSingleTap:(CGPoint)location
+{
+    NSLog(@"Did tap at %f %f", location.x, location.y);
+
+    // TODO: return physical pixels from API
+    location.x *= view.pixelScale;
+    location.y *= view.pixelScale;
+
+    [view setPosition:[view screenPositionToLngLat:location]];
+
+    [view pickFeaturesAt:location];
+}
+
+@end
+
 @interface MapViewController ()
 
 @end
 
 @implementation MapViewController
 
-- (void)viewDidLoad {
+- (void)viewDidLoad
+{
     [super viewDidLoad];
-    // Do any additional setup after loading the view, typically from a nib.
-}
 
+    self.mapViewDelegate = [[MapViewControllerDelegate alloc] init];
+    self.gestureDelegate = [[MapViewControllerRecognizerDelegate alloc] init];
 
-- (void)didReceiveMemoryWarning {
+    [super loadSceneFileAsync:@"https://tangrams.github.io/walkabout-style/walkabout-style.yaml"];
```

Comment:
I am not sure I have a good enough understanding to have an opinion on that topic. I don't believe it would pose an issue, but @blair1618 might have a better idea.